removeNthNodeFromEndOfList.cpp

```diff
@@ -0,0 +1,35 @@
+//time complexity: O(n)
+//space complexity:O(1)
+//approach: iterative
+//solved on leetcode? yes
+//problem faced? yes, giving run time error
+
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     ListNode *next;
+ *     ListNode() : val(0), next(nullptr) {}
+ *     ListNode(int x) : val(x), next(nullptr) {}
+ *     ListNode(int x, ListNode *next) : val(x), next(next) {}
+ * };
+ */
+class Solution {
+public:
+    ListNode* removeNthFromEnd(ListNode* head, int n) {
+        ListNode* dummy=new ListNode(0);
+        dummy->next=head;
+        if(head==NULL)
+        ListNode* slow=head;
+        ListNode* fast=head;
+        for(int i=1; i<=n+1; i++)
```

Comment:
if the given the n is 1, you are going over the given limit and fast will be at null.

Check this again